File: Personal_Web_API/Personal_Web_API/Dtos/SocialMedia.cs
namespace Personal_Web_API.Dtos
{
	public record GetSocialMedia (
		int Id,
		string SocialMediaName,
		string SocialMediaUrl
	);

	public class CreateSocialMedia
	{
		public string SocialMediaName { get; set; }
		public string SocialMediaUrl { get; set; }
		public int SocialMediaUserId { get; set; }
	}

	public record UpdateSocialMedia (
		int Id,
		string SocialMediaName,
		string SocialMediaUrl
	);
}

File: Personal_Web_UI/src/app/dtos/SocialMedia.ts
export class GetSocialMedia {
    id?: number;
    socialMediaName?: string;
    socialMediaUrl?: string;
}

export class CreateSocialMedia {
    socialMediaName?: string;
    socialMediaUrl?: string;
    
    constructor(socialMediaName?: string, socialMediaUrl?: string) {
        this.socialMediaName = socialMediaName;
        this.socialMediaUrl = socialMediaUrl;
    }
 }

 export class UpdateSocialMedia { 
    id?: number;
    socialMediaName?: string;
    socialMediaUrl?: string;

    constructor(id?: number, socialMediaName?: string, socialMediaUrl?: string) {
        this.id = id;
        this.socialMediaName = socialMediaName;
        this.socialMediaUrl = socialMediaUrl;
    }
  }